libraries/shared/storage/diff_repository_test.go

```diff
@@ -214,190 +215,145 @@ var _ = Describe("Storage diffs repository", func() {
-				ID:        rand.Int63(),
-				Status:    "new",
-				EthNodeID: db.NodeID,
-			}
-			insertTestDiff(unrecognizedPersistedDiff, db)
-
-			diffs, err := repo.GetUnrecognizedDiffs(0, 1)
-
-			Expect(err).NotTo(HaveOccurred())
-			Expect(diffs).To(BeEmpty())
-		})
-
-		It("does not send diffs that are marked as 'transformed'", func() {
-			transformedPersistedDiff := types.PersistedDiff{
-				RawDiff:   fakeStorageDiff,
-				ID:        rand.Int63(),
-				Status:    "transformed",
-				EthNodeID: db.NodeID,
-			}
-			insertTestDiff(transformedPersistedDiff, db)
-
-			diffs, err := repo.GetUnrecognizedDiffs(0, 1)
-
-			Expect(err).NotTo(HaveOccurred())
-			Expect(diffs).To(BeEmpty())
-		})
-
-		It("does not send diffs that are marked as 'noncanonical'", func() {
-			noncanonicalPersistedDiff := types.PersistedDiff{
-				RawDiff:   fakeStorageDiff,
-				ID:        rand.Int63(),
-				Status:    "noncanonical",
-				EthNodeID: db.NodeID,
-			}
-			insertTestDiff(noncanonicalPersistedDiff, db)
-
-			diffs, err := repo.GetUnrecognizedDiffs(0, 1)
-
-			Expect(err).NotTo(HaveOccurred())
-			Expect(diffs).To(BeEmpty())
-		})
-
-		It("enables seeking diffs with greater ID", func() {
-			blockZero := rand.Int()
-			for i := 0; i < 2; i++ {
-				fakeRawDiff := types.RawDiff{
-					Address:      test_data.FakeAddress(),
-					BlockHash:    test_data.FakeHash(),
-					BlockHeight:  blockZero + i,
-					StorageKey:   test_data.FakeHash(),
-					StorageValue: test_data.FakeHash(),
+				insertTestDiff(fakePersistedDiff, db)
+
+				diffs, err := repo.GetPendingDiffs(0, 1)
+				Expect(err).NotTo(HaveOccurred())
+				if present {
+					Expect(diffs).To(ConsistOf(fakePersistedDiff))
+				} else {
+					Expect(diffs).To(BeEmpty())
 				}
-				persistedDiff := types.PersistedDiff{
-					RawDiff:   fakeRawDiff,
+			},
+			KeepStatus(storage.Pending),
+			SkipStatus(storage.New),
+			SkipStatus(storage.Noncanonical),
+			SkipStatus(storage.Transformed),
+			SkipStatus(storage.Unrecognized),
+			SkipStatus(storage.Unwatched),
+		)
+
+		DescribeTable("only unrecognized",
+			func(status string, present bool) {
+				fakePersistedDiff := types.PersistedDiff{
+					RawDiff:   fakeStorageDiff,
 					ID:        rand.Int63(),
-					Status:    "new",
 					EthNodeID: db.NodeID,
+					Status:    status,
 				}
-				insertTestDiff(persistedDiff, db)
-			}
+				insertTestDiff(fakePersistedDiff, db)
+
+				diffs, err := repo.GetUnrecognizedDiffs(0, 1)
+				Expect(err).NotTo(HaveOccurred())
+				if present {
+					Expect(diffs).To(ConsistOf(fakePersistedDiff))
+				} else {
+					Expect(diffs).To(BeEmpty())
+				}
+			},
+			KeepStatus(storage.Unrecognized),
+			SkipStatus(storage.New),
+			SkipStatus(storage.Pending),
+			SkipStatus(storage.Noncanonical),
+			SkipStatus(storage.Transformed),
+			SkipStatus(storage.Unwatched),
+		)
+
+		// Surely there is a better way for me to setup these calls
+		type queryFunc func(int, int) ([]types.PersistedDiff, error)
```

Comment:
Gonna leave my own comment here - I'm 95% certain there's some way I can just pass repo.Func directly rather than creating these wrappers, but I was unable to figure it out. Any go-experts know that one?